services/mirror/mirror_pull_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+package mirror
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseRemoteUpdateOutput(t *testing.T) {
+	tests := []struct {
+		input   string
+		results []*mirrorSyncResult
+	}{
+		{
+			// create tag
+			input: "From https://xxx.com/xxx/xxx\n * [new tag]         v1.4       -> v1.4\n",
+			results: []*mirrorSyncResult{
+				{
+					refName:     "refs/tags/v1.4",
+					oldCommitID: gitShortEmptySha,
+					newCommitID: "",
+				},
+			},
+		},
+		{
+			// delete tag and create branch
+			input: "From https://xxx.com/xxx/xxx\n - [deleted]         (none)     -> v1.0.1\n * [new branch]      test/t3    -> test/t3\n * [new branch]      test/t4    -> test/t4\n",
+			results: []*mirrorSyncResult{
+				{
+					refName:     "v1.0.1",
+					oldCommitID: "",
+					newCommitID: gitShortEmptySha,
+				},
+				{
+					refName:     "refs/heads/test/t3",
+					oldCommitID: gitShortEmptySha,
+					newCommitID: "",
+				},
+				{
+					refName:     "refs/heads/test/t4",
+					oldCommitID: gitShortEmptySha,
+					newCommitID: "",
+				},
+			},
+		},
+		{
+			// delete branch
+			input: "From https://xxx.com/xxx/xxx\n - [deleted]         (none)     -> test/t2\n",
+			results: []*mirrorSyncResult{
+				{
+					refName:     "test/t2",
```

Comment:
Does it need prefix?

Reply:
Then I think it needs to be commented.

Reply:
I think I did similar things in my refname PR